zkSNACKs/WalletWasabi
# Change KeyPath for TestNet

#### This PR is still in progress, but testing is always appreciated.
Related to issue #6218 

## Description
When creating a _Wallet_, regardless of having a property that indicates if the wallet is used on **Main**- or **TestNet**, the `AccountKeyPath` for all wallets are set to be the same. `"84'/0'/0'"`, which is the default **KeyPath** for **MainNet**.
But as discussed in the issue above, this keeps the users from using _Trezor One_ on **TestNet**, because the Trezor rules are strict for BIP44, and will deny transactions and cause exception that has wrong **KeyPath**.

This PR should solve this issue. Altough **it won't touch older wallets** for the purpose of not risking to mess up files and the user's money, all wallets - including **hot** and _cold_ - that's generated afterwards on **TestNet** will have KeyPath set to `"84'/1'/0'"`, thus fixing the Trezor One issue on **TestNet**. 

## Tests

For the purpose of passing tests, I had to add a few things to the mocked hardware wallet tests, including:

- New ExtPubKeys for TestNet
- New argument checks for TestNet keypaths

## How to test

0. Remove `[yourHardwareWalletsName].json` file from the TestNet's Wallets folder if you have one.
1. Pull this issue
2. Run Wasabi
3. Load your Hardware Wallet and log in
4. Try to send some satoshis to a TestNet address
5. Confirm on hardware wallet

### +1 Internal test: 
Check your newly created wallet.json file to see if `"Network": "TestNet",` and `"AccountKeyPath":  "84'/1'/0'",`.
You can also create a new MainNet wallet and check if its the opposite. `"Network": "Main",` and `"AccountKeyPath":  "84'/0'/0'",`.
